pg_monz/usr-local-bin/pgsql_server_funcs.sh

```diff
@@ -11,19 +11,46 @@ PARAM1="$5"
 # Load the psql connection option parameters.
 source $PGSHELL_CONFDIR/pgsql_funcs.conf
 
+# Script required GNU sed which is not always the default so need to use gsed instead
+LOCALGSED="sed"
+gsed --version 2> /dev/null | grep -q "GNU"
+if [ $? -eq 0 ]; then
+	LOCALGSED="gsed"
+fi
+
 TIMESTAMP_QUERY='extract(epoch from now())::int'
 
-PGVERSION=$(psql -A -t -X -h $PGHOST -p $PGPORT -U $PGROLE $PGDATABASE -c 'select * from version()' | cut -d ' ' -f 2 | sed -n 's/^\([0-9]\+\(\.[0-9]\+\)\?\).*$/\1/p')
+PGVERSION=$(psql -A -t -X -h $PGHOST -p $PGPORT -U $PGROLE $PGDATABASE -c 'select * from version()' | cut -d ' ' -f 2 | $LOCALGSED -n 's/^\([0-9]\+\(\.[0-9]\+\)\?\).*$/\1/p')
```

Comment:
I think we can use `SELECT current_setting('server_version_num')` to find your PostgreSQL version.
It is much easier to compare.
Because it returns numeric values, we don't need use `bc` or `sed` commad.

```
PGVERSION=$(psql -A -t -X -h $PGHOST -p $PGPORT -U $PGROLE $PGDATABASE -c "SELECT current_setting('server_version_num')")

if [ $PGVERSION -ge 100000 ]; then
    ...
elif [ $PGVERSION -ge 90600 ]; then
    ...
else
    ...
fi
```

* PostgreSQL 9.4
```
$ psql -A -t -X -h 127.0.0.1 -p 11002 -U pengbo postgres -c "SELECT current_setting('server_version_num')"
90423
```
* PostgreSQL 12beta1
```
$ psql -A -t -X -h 127.0.0.1 -p 11002 -U pengbo postgres -c "SELECT current_setting('server_version_num')"
120000
```

Reply:
Nice finding I just test it on a 8.4.20 and it is also working.
80420